Hardware/Hardware/CardReader/CardReaderServiceError.swift

```diff
@@ -192,6 +192,19 @@ public enum UnderlyingError: Error {
     /// Catch-all error case. Indicates there is something wrong with the
     /// internal state of the CardReaderService.
     case internalServiceError
+
+    case locationIdMissing
+}
+
+extension UnderlyingError {
+    init(with configurationError: CardReaderConfigError) {
+        switch configurationError {
+        case .incompleteStoreAddress(_):
+            self =  .locationIdMissing
+        case .unknown(let error):
+            self  = UnderlyingError(with: error)
```

Comment:
It's convenient to fall back to the `+Stripe` extension here, which provides default behaviour of mapping errors it can't parse to `.internalServerError`... but it does seem like a bit of a side effect.

Two options come to mind:
1. Make this a failable init, and then at the call site use the Stripe init where it's appropriate.
2. Add a `.unknown` case to the `UnderlyingError` and use that, rather than always falling back to `.internalServerError`

I'm going to try option 1, because I think it'll be more explicit and help us see any consequences of the change.

Reply:
I would lean more towards option 2, to be honest. The way we would not need to deal with special cases at the call site

Reply:
To follow up from a slack chat; I gave the `.unknown` case a go here and it didn't really add anything. `.internalServiceError` is basically the unknown case already, it's intended as a catch all, and that's how we use it. 

It is quite close to `internalServ**er**Error`, which might lead to some confusion, so we could consider renaming it.